widgy_blog/admin.py

```diff
@@ -146,12 +146,15 @@ def save_model(self, request, obj, form, change):
             tags = layout_data.pop('tags', [])
             field = self.model._meta.get_field('content')
             obj.content = field.add_root(obj, layout_data)
-            obj.content.working_copy.content.tags = tags
+            obj.content.working_copy.content.tags.set(tags)
         else:
             # editing
             content = obj.content.working_copy.content
             for field_name, value in layout_data.items():
-                setattr(content, field_name, value)
+                if field_name == 'tags':
+                    content.tags.set(value)
```

Comment:
This is good, but should probably be generalized to work for all ManyToMany fields, not just `Tags`. Consider the case when the `BlogAdmin` is extended and uses another M2M field. You could use duck typing and just look for the `set` method, or you could do a `issubclass` check to see if you are using a `ManyRelatedManager`.